Zilon.Core/Zilon.Core/Tactics/StaticObject.cs

```diff
@@ -30,27 +32,24 @@ public StaticObject(IGraphNode node, int id)
         public bool IsMapBlock { get => GetIsMapBlock(); }
 
         /// <inheritdoc/>
-        public void AddModule<TSectorObjectModule>(TSectorObjectModule sectorObjectModule)
-        {
-            var sanitizedModuleType = GetSanitizedModuleType(typeof(TSectorObjectModule));
-            if (sanitizedModuleType is null)
-            {
-                throw new ArgumentException("Указанные объект не является модулем", nameof(sectorObjectModule));
-            }
+        public bool IsSightBlock { get => false; }
 
-            _modules.Add(sanitizedModuleType, sectorObjectModule);
+        /// <inheritdoc/>
+        public void AddModule<TStaticObjectModule>(TStaticObjectModule sectorObjectModule) where TStaticObjectModule: IStaticObjectModule
+        {
+            _modules.Add(sectorObjectModule.Key, sectorObjectModule);
         }
 
         /// <inheritdoc/>
-        public TSectorObjectModule GetModule<TSectorObjectModule>()
+        public TStaticObjectModule GetModule<TStaticObjectModule>(string key) where TStaticObjectModule : IStaticObjectModule
         {
-            return (TSectorObjectModule)_modules[typeof(TSectorObjectModule)];
+            return (TStaticObjectModule)_modules[key];
         }
 
         /// <inheritdoc/>
-        public bool HasModule<TSectorObjectModule>()
+        public bool HasModule<TStaticObjectModule>(string key) where TStaticObjectModule : IStaticObjectModule
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['TStaticObjectModule' is not used in the method.](https://app.codacy.com/manual/kreghek/Zilon_Roguelike/pullRequest?prid=5368096)